tests/DCPS/UnitTests/Serializer.cpp

```diff
@@ -0,0 +1,298 @@
+#include <dds/DCPS/Serializer.h>
+
+#include <gtest/gtest.h>
+
+using namespace OpenDDS::DCPS;
+
+TEST(serializer_test, Encoding__Encoding)
+{
+  Encoding enc;
+  EXPECT_EQ(0,enc.to_string().compare(0,24,"CDR/XCDR1, little-endian"));
```

Comment:
```suggestion
  EXPECT_EQ(0, enc.to_string().compare(0, 24,"CDR/XCDR1, little-endian"));
```

Maybe use the `EXPECT_STREQ` macro instead like you do latter in the file? I assume that gets better error reporting out of Google Test.

Reply:
So the reason I did it this way is because the string also contains "native" and "non-native". My feeling was that this could swap depending on the CI tests run but the endian prints wouldn't. If this is an incorrect assumption, I can definitely switch to the `EXPECT_STREQ` checks and check the full string output.

Reply:
Good point. It's unlikely and I'm unaware of any system we test on that is actually big endian, but it is still technically possible.

Reply:
Alternatively maybe change what string to expect based on the system byte over?

Reply:
Do we care that it correctly reflects native vs non-native? It's definitely more complete. Maybe good to know that it's correct to know that swapping is working correctly. I vote to leave as is but if you feel strongly, I can change it.

Reply:
I don't feel strongly about it either way, so it's fine as is.